exchanges/zb/zb_wrapper.go

```diff
@@ -699,7 +720,63 @@ func (z *ZB) ValidateCredentials() error {
 	return z.CheckTransientError(err)
 }
 
+// FormatExchangeKlineInterval returns Interval to exchange formatted string
+func (z *ZB) FormatExchangeKlineInterval(in kline.Interval) string {
+	switch in {
+	case kline.OneMin, kline.ThreeMin,
+		kline.FiveMin, kline.FifteenMin, kline.ThirtyMin:
+		return in.Short() + "in"
+	case kline.OneHour, kline.TwoHour, kline.FourHour, kline.SixHour, kline.TwelveHour:
+		return in.Short()[:len(in.Short())-1] + "hour"
+	case kline.OneDay:
+		return "1day"
+	case kline.ThreeDay:
+		return "3day"
+	case kline.OneWeek:
+		return in.Short()[:len(in.Short())-1] + "week"
+	}
+	return in.Short()
+}
+
 // GetHistoricCandles returns candles between a time period for a set time interval
-func (z *ZB) GetHistoricCandles(pair currency.Pair, a asset.Item, start, end time.Time, interval time.Duration) (kline.Item, error) {
-	return kline.Item{}, common.ErrNotYetImplemented
+func (z *ZB) GetHistoricCandles(pair currency.Pair, a asset.Item, start, end time.Time, interval kline.Interval) (kline.Item, error) {
+	if !z.KlineIntervalEnabled(interval) {
+		return kline.Item{}, kline.ErrorKline{
+			Interval: interval,
+		}
+	}
+
+	klineParams := KlinesRequestParams{
+		Type:   z.FormatExchangeKlineInterval(interval),
+		Symbol: z.FormatExchangeCurrency(pair, a).String(),
+	}
+
+	candles, err := z.GetSpotKline(klineParams)
+	if err != nil {
+		return kline.Item{}, err
+	}
+
+	ret := kline.Item{
+		Exchange: z.Name,
+		Pair:     pair,
+		Asset:    a,
+		Interval: interval,
+	}
+
+	for x := range candles.Data {
+		ret.Candles = append(ret.Candles, kline.Candle{
+			Time:   candles.Data[x].KlineTime,
+			Open:   candles.Data[x].Open,
+			High:   candles.Data[x].Close,
+			Low:    candles.Data[x].Low,
+			Close:  candles.Data[x].Close,
+			Volume: candles.Data[x].Volume,
+		})
+	}
+	return ret, nil
+}
+
+// GetHistoricCandlesEx returns candles between a time period for a set time interval
+func (z *ZB) GetHistoricCandlesEx(p currency.Pair, a asset.Item, start, end time.Time, interval kline.Interval) (kline.Item, error) {
+	return z.GetHistoricCandles(p, a, start, end, interval)
```

Comment:
Start and end is unused in `GetHistoricCandles`. How's this for fun? This function returns `zb GetSpotKline rawKlines is nil` using the same parameters as `GetHistoricCandles`, but calling `GetHistoricCandles` returns data 🎉 

Whatever the case, since ZB uses a `since` parameter for ObtainKLine endpoint, the startdate should at least be used or you need to filter out data yourself in this function if you're not going to use it in `GetHistoricCandles`.

Reply:
the rawKlines is most times due to rate limiting i will have a look at the numbers again they say 1 per 1 second but i have found its more 1 per 1.7 seconds might need to bump it up to 1 per 2 seconds 

when i was testing zb the since param returned inconsistent data but i will add support back for it

Reply:
When you get time can I get you to test this again with verbose on as well see if its returning a rate limit response still?

Reply:
If I remove all other funcs from exchange_wrapper_issues, I don't get rate limited with this. However, it contains data outside the range provided, `2020-05-11` but goes back to `2017-10-31` which is probably why you get rate limited